packages/medusa/src/repositories/money-amount.ts

```diff
@@ -52,35 +52,26 @@ export class MoneyAmountRepository extends Repository<MoneyAmount> {
     variantId: string,
     prices: Price[]
   ): Promise<void> {
-    const existingPricesIdsQuery = this.createQueryBuilder()
-      .select("id")
-      .where({
-        variant_id: variantId,
-        price_list_id: IsNull(),
-      })
-      .andWhere(
-        new Brackets((qb) => {
-          prices.forEach((price) => {
-            qb.orWhere(
-              new Brackets((qb) => {
-                qb.where({
-                  currency_code: price.currency_code,
-                  region_id: IsNull(),
-                }).orWhere({ region_id: price.region_id })
-              })
-            )
-          })
-        })
-      )
+    const currencyCodes = prices.map((p) => p.currency_code).filter((v) => v)
+    const regionIds = prices.map((p) => p.region_id).filter((v) => v)
+
+    const constraints = {
+      variant_id: variantId,
+      price_list_id: IsNull(),
+    }
 
     await this.createQueryBuilder()
       .delete()
-      .where({
-        variant_id: variantId,
-        price_list_id: IsNull(),
-      })
-      .andWhere(`id NOT IN (${existingPricesIdsQuery.getQuery()})`)
-      .setParameters(existingPricesIdsQuery.getParameters())
+      .where([
+        {
+          ...constraints,
+          currency_code: Not(In(currencyCodes)),
```

Comment:
question: According to the previous query, we should add `region_id: IsNull()` as a constraint here. Should that still be the case? cc @kasperkristensen

Reply:
@kasperkristensen I believe that there is an issue, 

we update a variant with the following prices:
- USD
- USD / test-region
- eur

Then we update the variant with
- USD
- EUR

The delete constraint check `not(in(["usd", "eur"]))` but since `USD / test-region` is USD, this one will not get deleted right?

Reply:
I think that you are right @olivermrbl , the region is null needs to be added to first constraint 💪

Reply:
I have rewrite the constraints to match the opposite of the previous select/delete not in